.bazelci/presubmit.yml

```diff
@@ -109,7 +109,53 @@ tasks:
       # - "--linkopt=-fuse-ld=lld"
     build_targets: *default_linux_targets
     test_targets: *default_linux_targets
-
+  crate_universe_examples_ubuntu2004:
+    name: Crate Universe Examples
+    platform: ubuntu2004
+    environment:
+      RULES_RUST_CRATE_UNIVERSE_BOOTSTRAP: true
+    working_directory: examples/crate_universe
+    build_targets:
+      - "//..."
+    test_targets:
+      - "//..."
+  crate_universe_rbe_ubuntu1604:
+    name: Crate Universe Examples
+    platform: rbe_ubuntu1604
+    environment:
+      RULES_RUST_CRATE_UNIVERSE_BOOTSTRAP: true
```

Comment:
We should probably remove the checked in lockfile for the `uses_sys_crate` example.

Right now, for a `crate_universe_resolver` binary built in release mode (as the bootstrapper does), the `crate_universe_resolver` version number is what's used in the cache key hash to factor in the resolver's logic. Accordingly, we should either use a unique version number per CI run (e.g. by appending `-${git_hash}` to the version number in the `Cargo.toml`), or we should force the example to regenerate its lockfile every time by setting `RULES_RUST_REPIN=true` for these CI jobs.

But if nothing is going to be automatically updating the lockfile when we change things, it's probably best to just delete it for now, while things are in flux.

Reply:
I don't think we should be regenerating a lockfile and I don't think it's the best idea to bake the version of the resolver into the lockfile. I feel the lockfile should be probably have it's own version and only be updated when changes are made to the resolver which would actually impact the lockfile. But for the purposes of this PR, I'd like to have something which uses the lockfile but doesn't update it. This way we can identify drift in the structure of the lockfiles and keep CI more deterministic.

That said, what do you think I should do?

Reply:
> I feel the lockfile should be probably have it's own version and only be updated when changes are made to the resolver which would actually impact the lockfile.

I agree with this :)

> But for the purposes of this PR, I'd like to have something which uses the lockfile but doesn't update it.

I think in practice that means every PR which modifies `crate_universe` will need to bump the version number in the `Cargo.toml` of the resolver, and re-generate the lockfile. Which feels like a bad experience, which is why I'm suggesting deleting it for now.

Right now things happen to be green because the checked in lockfile happens to be compatible with the rust code, but I imagine that will stop being the case almost immediately, and the resolver will try to use incompatible values and it will be annoying.

> This way we can identify drift in the structure of the lockfiles and keep CI more deterministic.

I like these as goals, but I don't think the lockfile stuff is mature enough for it to actually work out in practice yet.

Reply:
Removed the lockfiles